--- a/sway/desktop/layer_shell.c
+++ b/sway/desktop/layer_shell.c
@@ -132,7 +132,7 @@
 			seat_set_focus_layer(seat, topmost->layer_surface);
 		} else if (seat->focused_layer &&
 				seat->focused_layer->current.keyboard_interactive
-					!= ZWLR_LAYER_SURFACE_V1_KEYBOARD_INTERACTIVITY_EXCLUSIVE) {
+					== ZWLR_LAYER_SURFACE_V1_KEYBOARD_INTERACTIVITY_NONE) {
 			seat_set_focus_layer(seat, NULL);
 		}
 	}
